generator/sbpg/targets/rust.py

```diff
@@ -101,17 +134,69 @@ def parse_type(field):
     # This is an inner class, call default constructor
     return "%s::parse(_buf)" % field.type_id
 
+def gps_time(msg, all_messages):
+    def time_aware_header(type_id):
+        for m in all_messages:
+            if m.identifier == type_id:
+                return any([f.identifier == "t" for f in m.fields])
+        return False
+
+    def gen_body():
+        header = False
+        tow = False
+        wn = False
+
+        for f in msg.fields:
+            if f.identifier == "header" and time_aware_header(f.type_id):
+                header = True
+            elif f.identifier == "tow":
+                assert f.units == "ms"
+                tow = True
+            elif f.identifier == "wn":
+                wn = True
+
+        if header:
+            return GPS_TIME_HEADER
+        elif tow and wn:
+            return GPS_TIME
+        elif tow:
+            return GPS_TIME_ONLY_TOW
+        else:
+            return None
+
+    def gen_ret():
+        name = "Base" if msg.identifier in BASE_TIME_MSGS else "Rover"
+        return f"Some(Ok(crate::time::MessageTime::{name}(gps_time.into())))"
+
+    if msg.identifier in SKIP_GPS_TIME_MSGS:
+        return ""
+
+    body = gen_body()
+    if body is None:
+        return ""
+
+    ret = gen_ret()
+
+    return f"""\
```

Comment:
line continuations are an abomination that cause problems for tools parsing the python.  Doing `var = """...""".strip()` usually means it is unnecessary.

also the code is prettier if `textwrap.dedent` is used, and usually .

Reply:
I think you missed this one, and did all the others.

Reply:
good catch, thanks